### ISSUE TITLE
Issue on reproduce MuJoCo results-HalfCheetah-v2

### ISSUE THREAD

**Issue by shuoye1000** [Div99/IQ-Learn#5]
Dear Author, it's an honor to see your paper and code! I am a novice in this area and now I am trying to reproduce the effect of your experiment, but I have encountered some obstacles. In Half-Cheetah, I don't get the 5076.6 points in the paper, even my reward is less than 0 in most cases, the code is not modified, is the reason the hyperparameter setting? If so, could you share your hyperparameter setting? Thanks for sharing!

**Comment by Div-Infinity** [comment]
Hi, please set `agent.learn_temp=False` and it should work. In my experience, SAC style temperature learning is not very stable with IQ-Learn and can prevent the method from converging. I pushed a change where the temperature learning is disabled by default.

**Comment by Div-Infinity** [comment]
Here are some training results I ran yesterday with 1 and 10 expert demos, with the hyperparams in `run_mujoco.sh` with the temp learning disabled:

![W B Chart 4_14_2022, 11_04_56 AM](https://user-images.githubusercontent.com/22247549/163454173-a61c17e8-a6af-4c09-ad37-38f3ab407005.png)

**Comment by Div-Infinity** [comment]
**Imp Update**: I found that temperature learning can also work well if the [SAC target_entropy param](https://github.com/Div99/IQ-Learn/blob/aeca377fff6e8880adf2f1ba946ee42d21b6e90b/iq_learn/agent/sac.py#L36) is set to much lower than −dim(A), as we don't need much exploration for IL settings. For Half-Cheetah, setting the **target_entropy=-24** works well. So empirically **4 * -dim(A)** could be a good value to try with IQ-Learn

Reward Curves:
![W B Chart 4_14_2022, 5_09_19 PM](https://user-images.githubusercontent.com/22247549/163496223-343c172d-43fb-4e99-b8b8-1b44564e4493.png)

**Comment by shuoye1000** [comment]
关于您给出的建议，我重新下载了您的代码并且重新运行了脚本，这是我使用的命令：【python train_iq.py env=cheetah agent=sac expert.demos=10 method.loss=value method.regularize=True agent.actor_lr=3e-05 seed=0 agent.learn_temp=False】
This is the output of my console parameters.
![image](https://user-images.githubusercontent.com/37900782/164141245-293f607c-f605-4846-bada-9e49e0ede3bb.png)
![image](https://user-images.githubusercontent.com/37900782/164141328-902327bb-144f-4722-9a9d-ed8b8d4c298b.png)
I'm too clumsy, whether the   obs_dim and  action_dim  of the agent need to be modified by myself ？
And here are the training results I've gotten so far：
![image](https://user-images.githubusercontent.com/37900782/164141665-0fe062e8-371d-4877-93db-dc74ff907600.png)
I would be very grateful for your guidance!

**Comment by Div-Infinity** [comment]
Hi, the obs_dim and action_dim are set automatically by the code and you can ignore them. The default SAC agent temperature in the project repo was too high and I pushed a fix setting it to `1e-2` . For the HalfCheetah environment, both 1e-2 or 1e-3 temperature values should work very well.

**Comment by shuoye1000** [comment]
Through your tireless guidance, I finally solved the problem, thank you very much for your patience! Other than that, I still found some other minor issues, it seems that the following are missing from our code: Humanoid experiment data (parameters, expert data, etc.), Breakout expert data, Space Invaders expert data. If you could upload these data, it would help me to reproduce your paper better. And if the file is too large, zip seems to be a good option. You are a very charismatic author, good luck with your work!
![O`BH@X8)JX}IB06ZR`%ODD2](https://user-images.githubusercontent.com/37900782/164626068-31471437-d420-439b-8ad9-0a01405e0f4e.gif)

**Comment by Div-Infinity** [comment]
Hi, the link to the Atari datasets is fixed. For Humanoid, the expert data got accidentally deleted, and I will try to retrain a SAC agent if I can find time on the weekend to collect new expert trajectories.

**Comment by liyc-ai** [comment]
Hi, thanks to your work and patient guidance. Just a small question, why dose your SAC on HalfCheetah-v2 only get 5000 points? From my own experience, SAC can reach to at least 12000 points within 1M steps. See similar results from [OpenAI's SpinningUp ](https://spinningup.openai.com/en/latest/spinningup/bench.html) or the picture below.
![image](https://user-images.githubusercontent.com/38598166/166450100-d2d17f33-652b-4849-9e11-905505907d77.png)

**Comment by liushunyu** [comment]
> Hi, the link to the Atari datasets is fixed. For Humanoid, the expert data got accidentally deleted, and I will try to retrain a SAC agent if I can find time on the weekend to collect new expert trajectories.

HI, I cannot find Space Invaders expert data in the link

**Comment by Div-Infinity** [comment]
For the half-cheetah expert, we may have trained with a single critic, instead of double critics so performance is not as high. (I don't know exactly why we did it, but the baselines we compared against like ValueDICE had similar expert performance so we likely didn't optimize too much). Nevertheless, a better expert should easily translate to better imitation with IQ-Learn, as it easily saturates the current expert performance at ~5000 level

**Comment by Div-Infinity** [comment]
> HI, I cannot find Space Invaders expert data in the link

Sorry for missing SpaceInvaders in the dataset release, I will add it to the GDrive.

**Comment by liushunyu** [comment]
> > HI, I cannot find Space Invaders expert data in the link
> 
> Sorry for missing SpaceInvaders in the dataset release, I will add it to the GDrive.

Thanks!!! By the way, can I run SQIL based on your code?  I find the "sqil.yaml" in the "iq_learn/conf/method". How can I run it? Moreover, do you provide the GAIL implementation using the same expert data as you?

**Comment by Div-Infinity** [comment]
We used this repo for running GAIL: https://github.com/ikostrikov/pytorch-a2c-ppo-acktr-gail. You can add our data-loading setup to it to use our provided expert demos.  SQIL code was removed in the newer commits from our repo but is very simple to implement, and you can check an old version of the codebase

**Comment by Div-Infinity** [comment]
I have added the expert datasets along with IQ-Learn results on Humanoid-v2 along. Also released a script to generate your expert trajectories for new environments
